mesonbuild/cmake/tracetargets.py

```diff
@@ -137,9 +137,9 @@ def resolve_cmake_trace_targets(target_name: str,
         elif 'IMPORTED_IMPLIB' in tgt.properties:
             res.libraries += [x for x in tgt.properties['IMPORTED_IMPLIB'] if x]
         elif f'IMPORTED_LOCATION_{cfg}' in tgt.properties:
-            res.libraries += [x for x in tgt.properties[f'IMPORTED_LOCATION_{cfg}'] if x]
+            targets += [x for x in tgt.properties[f'IMPORTED_LOCATION_{cfg}'] if x]
         elif 'IMPORTED_LOCATION' in tgt.properties:
-            res.libraries += [x for x in tgt.properties['IMPORTED_LOCATION'] if x]
+            targets += [x for x in tgt.properties['IMPORTED_LOCATION'] if x]
```

Comment:
Possible downside: 
A missing lib will be turned into a meson-time warning instead of a link-time hard error.

The alternative is to completely remove framework recasting from this loop (single responsibility: resolve *targets*), and do that in a separate function to be applied on the resolved list.

Reply:
Maybe I was still a litte bit nervous at the time of writing because some of my tests suffer from not being able to resolve imported targets from transitive usage requirements. In building other people's packages, I would prefer a meson-time *error* about `ZLIB::ZLIB` or `CoreFoundation` being unknown, instead of a link time error about some missing symbols of unknown origin.
(As a coincidence, that's the benefit of `$<TARGET_NAME:literal>`: It is known at cmake-generation time that `literal` is meant to be a target, not something to be resolved by the linker.)

Reply:
`meson setup --fatal-meson-warnings` would fatally error out with "Dependency ZLIB::ZLIB for mycoolproject was not found", although it does require a generally warning-free configure step.

I don't fully understand when it's a good idea for this to be a warning vs an error...

Reply:
> I don't fully understand when it's a good idea for this to be a warning vs an error...

I do not have a general answer either. 

But for the particular case of a member of the link interface of a CMake target to be known to be a CMake target (either using a `namespace::` or the generator expression), it should be an error if the target isn't available when resolving targets IMO.

Reply:
> The alternative is to completely remove framework recasting from this loop (single responsibility: resolve _targets_), and do that in a separate function to be applied on the resolved list.

Would you like to see this alternative as another commit or in a separate PR? (Still to be done.)